src/OpenTelemetry.Contrib.Instrumentation.Azure/Implementation/AzureSdkDiagnosticListener.cs

```diff
@@ -74,35 +74,18 @@ public override void OnStartActivity(Activity activity, object valueValue)
                 operationName = this.GetOperationName(activity);
             }
 
-            List<ActivityLink> links = null;
-            if (this.linksPropertyFetcher.Fetch(valueValue) is IEnumerable<Activity> activityLinks)
-            {
-                if (activityLinks.Any())
-                {
-                    links = new List<ActivityLink>();
-                    foreach (var link in activityLinks)
-                    {
-                        if (link != null)
-                        {
-                            links.Add(new ActivityLink(new ActivityContext(link.TraceId, link.ParentSpanId, link.ActivityTraceFlags)));
-                        }
-                    }
-                }
-            }
-
-            // Ignore the activity and create a new one using ActivitySource.
-            // The new one will have Sampling decision made using extracted Links as well.
-            AzureSDKActivitySource.StartActivity(operationName, activityKind, activity.Id, activity.TagObjects, links);
+            activity.DisplayName = operationName;
+            this.activitySource.Start(activity, activityKind);
         }
 
-        public override void OnStopActivity(Activity current, object valueValue)
+        public override void OnStopActivity(Activity activity, object payload)
         {
-            // nothing to be done.
+            this.activitySource.Stop(activity);
         }
 
-        public override void OnException(Activity activity, object valueValue)
+        public override void OnException(Activity activity, object payload)
         {
-            activity.SetStatus(Status.Error.WithDescription(valueValue?.ToString()));
+            activity.SetStatus(Status.Error.WithDescription(payload?.ToString()));
```

Comment:
shouldn't we get only the exception message here?

For aspnetcore I did:
```csharp
if (!this.stopExceptionFetcher.TryFetch(payload, out Exception exc) || exc == null)
{
     AspNetCoreInstrumentationEventSource.Log.NullPayload(nameof(HttpInListener), nameof(this.OnException));
     return;
}
activity.RecordException(exc);
activity.SetStatus(Status.Error.WithDescription(exc.Message));
```

Reply:
The changes to this specific line of code is not a part of this PR. Just renamed the variable name. Yes we should change it as per ASP.NET Core, but wanted to keep this PR very simple and will get this addressed later.